config/config.go

```diff
@@ -476,6 +463,26 @@ func supportedFormat(format string) bool {
 	return false
 }
 
+func processUnusedConfigKeys(md mapstructure.Metadata, file string) error {
```

Comment:
Erroring on unused configuration values is a great idea.

Reply:
Is there a test for this?

Reply:
The config decoder had [already supported erroring unused configuration](https://github.com/hashicorp/consul-terraform-sync/pull/169/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R347
) and is [tested here](https://github.com/hashicorp/consul-terraform-sync/blob/master/config/config_test.go#L136-L149)

However with the way the decoder works, if that is set to `true`, it doesn't populate the metadata to figure out which keys are unused and returns an error. Another approach could string match that error for `'provider'` for this special case handling. So I swapped it to false and manually processed the unused keys.